FIAT/check_format_variant.py

```diff
@@ -8,14 +8,15 @@ def check_format_variant(variant, degree):
     match = re.match(r"^integral(?:\((\d+)\))?$", variant)
     if match:
         variant = "integral"
-        quad_degree, = match.groups()
-        quad_degree = int(quad_degree) if quad_degree is not None else (degree + 1)
-        if quad_degree < degree + 1:
```

Comment:
It's good to get rid of this logic since the variant can't know what the minimal degree has to be for all elements?

Reply:
We have DOFs as moments against degree k, k-1, k-2, and k-3 polynomials. So the minimal degree for the quadraturee is the embedded degree of the FE space plus the degree of the polynomials against which we integrate. Here I only use the variant to define the degree of the argument to the dofs (the interopolate?). Maybe we should pick a better name for `interpolant_deg`?

Reply:
It plays two roles: "amount we bump up over the minimal quadrature degree", or also "the degree of polynomial input on which the quadrature approximation to the node will be exact.  I'm not sure what the best name for it i.

My comment was intended to say that, absent information about the definition of each element, our variant selector can't tell what the minimally acceptable degree of quadrature is as a function of the element degree.  Having `check_form_variant` work with an "extra degree" (or our other interpretation) is meaningful absent the context of what an element's dofs actually are.